include/random_walks/new_gaussian_accelerated_billiard_walk.hpp

```diff
@@ -0,0 +1,274 @@
+            :   param(0, false)
+    {}
+
+    struct parameters
+    {
+        parameters(double L, bool set)
+                :   m_L(L), set_L(set)
+        {}
+        double m_L;
+        bool set_L;
+    };
+
+    struct update_parameters
+    {
+        update_parameters()
+                :   facet_prev(0), hit_ball(false), inner_vi_ak(0.0), ball_inner_norm(0.0)
+        {}
+        int facet_prev;
+        bool hit_ball;
+        double inner_vi_ak;
+        double ball_inner_norm;
+    };
+
+    parameters param;
+
+
+    template
+    <
+            typename Polytope,
+            typename RandomNumberGenerator
+    >
+    struct Walk
+    {
+        typedef typename Polytope::PointType Point;
+        typedef typename Polytope::MT MT;
+        typedef typename Polytope::VT VT;
+        typedef typename Point::FT NT;
+
+        template <typename GenericPolytope>
+        Walk(GenericPolytope& P,
+             Point const& p,
+             MT const& E,   // covariance matrix representing the Gaussian distribution
+             RandomNumberGenerator &rng)
+        {
+            _update_parameters = update_parameters();
+            _L = compute_diameter<GenericPolytope>::template compute<NT>(P);
+
+            _AA.noalias() = P.get_mat() * P.get_mat().transpose();
+            _rho = 1000 * P.dimension(); // upper bound for the number of reflections (experimental)
+            initialize(P, p, E, rng);
+        }
+
+        template <typename GenericPolytope>
+        Walk(GenericPolytope& P,
+             Point const& p,
+             MT const& E,   // covariance matrix representing the Gaussian distribution
+             RandomNumberGenerator &rng,
+             parameters const& params)
+        {
+            _update_parameters = update_parameters();
+            _L = params.set_L ? params.m_L
+                              : compute_diameter<GenericPolytope>
+                                ::template compute<NT>(P);
+            
+            _AA.noalias() = P.get_mat() * P.get_mat().transpose();
+            _rho = 1000 * P.dimension(); // upper bound for the number of reflections (experimental)
+            initialize(P, p, E, rng);
+        }
+
+        template <typename GenericPolytope>
+        inline void apply(GenericPolytope& P,
+                          Point &p,       // a point to return the result
+                          MT const& E,   // covariance matrix representing the Gaussian distribution
+                          unsigned int const& walk_length,
+                          RandomNumberGenerator &rng)
+        {
+            unsigned int n = P.dimension();
+            NT T;
+            const NT dl = 0.995;
+            int it;
+            NT coef = 1.0;
+            NT vEv;
+
+            for (auto j=0u; j<walk_length; ++j)
+            {
+                T = -std::log(rng.sample_urdist()) * _L;
+
+
+                Eigen::LLT<MT> lltOfE(E.inverse()); // compute the Cholesky decomposition of inv(E)
```

Comment:
Why do we need to compute the cholesky decomposition of E in every iteration since it should be considered fixed?

We could compute _L_cov in the constructor and initialize a member variable _E. Then, E shouldn't be an input in `apply()` neither in `initialize()`.